docs/usecase.rst

```diff
@@ -21,12 +21,47 @@ Scripting user
 
 Python code written directly against the data flow client library.
 
+.. seealso:: :doc:`dataflow`
+
+Run a sequence of dependent tasks
+---------------------------------
+
+| 1. Import data flow client package.
+| 2. Add an operation to the work graph.
+| 3. Add a dependent operation to the work graph.
+| 4. Request results.
+|  4.1 Work is implicitly dispatched.
+|  4.2 Interpreter blocks until results are available.
```

Comment:
I think a tacit assumption here is that the resource on which this work will be executed have been already acquired and are available for work execution? The same assumption holds also for setting up the environment required by the execution (would it be having a worker up and running or a set of libraries and variables set)?

Reply:
Computing resources should not be acquired until as late as possible. The script may have already executed to completion, in which case the final results are already locally available. Also, as much of the graph should be declared as possible before resources are acquired so that appropriate resources may be requested.

Reply:
Resources here means a certain amount of cores/GPU held by a certain amount of worknodes on one or more HPC machines. Analogously, resource here could also mean some virtual machines on a cloud infrastructures, even if we decided to start with HPC-based resources. Thus, acquiring resources means: submitting one or more jobs to an HPC machine. Once these resources have been acquired and are made available, then they might be used to execute some unit of work. I agree that in this case, the binding between the unit of work and the available resources should happen as late as possible. If you did not acquire resources, how the script can be executed? On what resources?